packages/shared/src/hooks/usePluginHostPermission.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react'
+import { useAsyncFn, useAsyncRetry } from 'react-use'
+import type { Plugin } from '@masknet/plugin-infra'
+import { MaskMessages } from '@masknet/shared-base'
+import { useSNSAdaptorContext } from '@masknet/plugin-infra/content-script'
+
+export function usePluginHostPermissionCheck(plugins: Plugin.Shared.Definition[]) {
```

Comment:
why do you need to move it to the shared folder? app should not use it.

Reply:
need